addons/mass_mailing/static/src/components/mailing_test_popover/mailing_test_popover.js

```diff
@@ -0,0 +1,57 @@
+import { parseEmail } from "@mail/utils/common/format";
+import { useService } from "@web/core/utils/hooks";
+import { isEmail } from "@web/core/utils/strings";
+
+import { Component, onWillStart, useState } from "@odoo/owl";
+
+export class MailingTestPopover extends Component {
```

Comment:
This part needs to be completely re-done.

We are adding layers of JavaScript components to basically have the same behavior as mailing.mailing.test but with a slightly different layout.

To make matters worse, we're keeping "mailing.mailing.test" as a Wizard but don't use it anymore as such, it's just some kind of proxy to compute default values and send the test emails.

Would say we scrap all that and just keep mailing.mailing.test as currently and call it from the mailing.mailing.preview, I know it's a "modal on top of a modal" but the slight UI drawback is not worth this overhead and lack of proper model design.